search_params.py

```diff
@@ -0,0 +1,132 @@
+import argparse
+import logging
+import os
+import time
+import yaml
+from datetime import datetime
+from pathlib import Path
+
+from ray import tune
+
+from libmultilabel import data_utils
+from libmultilabel.evaluate import evaluate
+from libmultilabel.model import Model
+from libmultilabel.utils import ArgDict, set_seed, init_device
+
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s %(levelname)s:%(message)s')
+
+
+def training_function(config):
+    model_config = ArgDict(config)
+    datasets = data_utils.load_datasets(model_config)
+    word_dict = data_utils.load_or_build_text_dict(model_config, datasets['train'])
+    classes = data_utils.load_or_build_label(model_config, datasets)
+
+    model = Model(model_config, word_dict, classes)
+    model.train(datasets['train'], datasets['val'])
+    model.load_best()
+
+    # return best eval metric
+    dev_loader = data_utils.get_dataset_loader(model_config, datasets['val'], model.word_dict, model.classes, train=False)
+    results = evaluate(model_config, model, dev_loader, split='val', dump=True)
+    yield results
+
+
+def init_model_config(config_path):
+    with open(config_path) as fp:
+        args = yaml.load(fp, Loader=yaml.SafeLoader)
+
+    # set relative path to absolute path (_path, _file, _dir)
+    for k, v in args.items():
+        if isinstance(v, str) and (os.path.isfile(v) or os.path.isdir(v)):
+            args[k] = os.path.abspath(v)
+
+    model_config = ArgDict(args)
+    set_seed(seed=model_config.seed)
+    model_config.device = init_device(model_config.cpu)
+    model_config.run_name = '{}_{}_{}'.format(
+        model_config.data_name,
+        Path(model_config.config).stem if model_config.config else model_config.model_name,
+        datetime.now().strftime('%Y%m%d%H%M%S'),
+    )
+    logging.info(f'Run name: {model_config.run_name}')
+    return model_config
+
+
+def init_search_space(search_func, values):
+    """Initialize the search space by the given search algorithm.
+    Currently, the search algorithm decides what search function is used for all parameters.
+    """
+    if search_func == 'grid':
+        return tune.grid_search(values)
+    elif search_func == 'uniform':
+        assert len(values) == 2 and values[0] < values[1]
+        # sample an option uniformly between values[0] and values[1]
+        return tune.uniform(values[0], values[1])
+    else:
+        # sample an option uniformly from list of values
+        return tune.choice(values)
+
+
+def get_search_algorithm(search_alg, metric=None, mode=None):
+    """Specify a search algorithm. You should pip install this search algorithm first.
+    See more details here: https://docs.ray.io/en/master/tune/api_docs/suggestion.html
```

Comment:
So users need to check this address for available algs? If so, should mention it in README